dali/sasha/saverify.cpp

```diff
@@ -199,13 +200,22 @@ class CFileCrcItem: public CInterface
 class CFileCrcList
 {
     CIpTable dafilesrvips;
+    Owned<IUserDescriptor> udesc;
 public:
     bool &stopped;
     CIArrayOf<CFileCrcItem> list;
 
     CFileCrcList(bool &_stopped)
         : stopped(_stopped)
     {
+        StringBuffer userName;
+        serverConfig->getProp("@SashaUser", userName);
+        StringBuffer password;
+        serverConfig->getProp("@SashaUserPassword", password);
```

Comment:
I don't think we want to store passwords in the configuration files.  I think they can probably be read from the eclwatch page, and the decryption function is freely available.  Should discuss with @afishbeck on the best approach.

Reply:
We are definitely trying to move away from this and would rather not add new cases.

In the cloud it's fairly clear, we would:
1. only let internal components connect to DALI.
2. When we can't automatically trust where the component is coming from use certificates or keys to establish trust.
3. Where we do need passwords or keys for things it would be stored as kubernetes secrets, or in a vault.

For bare metal it's not as clear.

Do DALI restrictions limit access to known endpoints?  If so then perhaps just the component/instance name without pw is enough?

If we need pw we should start treating them like secrets and storing them outside the config file where only hpcc user can access them.  

Longer term when we add vault support we can support it in bare metal as well.

Reply:
Yes, I was thinking about it some more, and I don't think the password is needed.  Nothing else passes the password to lookup any more (since it isn't even stored), so there is no need (or benefit) for sasha to.